scripts/pylib/display-twister-harness/camera_shield/README.rst

```diff
@@ -0,0 +1,119 @@
+==============
+Display capture Twister harness
+==============
+
+
+Configuration example
+---------------------
+
+```yaml: config.yaml
+case_config:
+  device_id: 0  # Try different camera indices
+  rex_x: 1280   # x reslution
+  res_y: 720    # y resolution
+  fps: 30       # analysis frame pre-second
+  run_time: 20  # Run for 20 seconds
+tests:
+  timeout: 30   # second wait for prompt string
+  prompt: "screen starts" # prompt to show the test start
+  expect: ["sample.display.shield"]
+plugins:
+  - name: signature
+    module: plugins.signature_plugin
+    class: VideoSignaturePlugin
+    status: "enable"
+    config:
+      operations: "compare" # operation ('generate', 'compare')
+      metadata:
+        name: "sample.display.shield" # finger-print stored metadata
+        platform: "frdm_mcxn947"
+      directory: "./fingerprints" # fingerprints directory to compare with not used in generate mode
+      duration: 100 # number of frames to check
+      method: "combined" #Signature method ('phash', 'dhash', 'histogram', 'combined')
+      threshold: 0.65
+      phash_weight: 0.35
+      dhash_weight: 0.25
+      histogram_weight: 0.2
+      edge_ratio_weight: 0.1
+      gradient_hist_weight: 0.1
+```
+
+example zephyr display tests
+----------------------------
+
+1. Setup camera to capture display content
+
+ - uvc compatible camera with 200M resolution
+ - a light-blocking black curtain
+ - a PC host where camera connect to
+ - DUT connect to same PC host for flashing and console uart
+
+2. Generate video fingerprints
```

Comment:
If I understand correctly, first video fingerprints of a known working version are generated: "reference fingerprints"
Then a new firmware to test is loaded.
Then new "test fingerprints" are generated during the tests and compared with the "reference fingerprints".

If so, there seems like a missing step where the reference firmware known to work is loaded inserted on the first run to generate the reference fingerprints.

Reply:
you are right, let me add